libfreerdp/gdi/graphics.c

```diff
@@ -339,43 +337,66 @@ static BOOL gdi_Glyph_BeginDraw(rdpContext* context, UINT32 x, UINT32 y,
 	if (!gdi->drawing || !gdi->drawing->hdc)
 		return FALSE;
 
-	if (!gdi_decode_color(gdi, bgcolor, &bgcolor, NULL))
-		return FALSE;
+	return gdi_SetClipRgn(gdi->drawing->hdc, x, y, width, height);
+}
+
+static BOOL gdi_Glyph_BeginDraw(rdpContext* context, INT32 x, INT32 y,
+                                INT32 width, INT32 height, UINT32 bgcolor,
+                                UINT32 fgcolor, BOOL fOpRedundant)
+{
+	rdpGdi* gdi;
 
-	if (!gdi_decode_color(gdi, fgcolor, &fgcolor, NULL))
+	if (!context || !context->gdi)
 		return FALSE;
 
-	gdi_SetTextColor(gdi->drawing->hdc, bgcolor);
-	gdi_SetBkColor(gdi->drawing->hdc, fgcolor);
+	gdi = context->gdi;
+
+	if (!gdi->drawing || !gdi->drawing->hdc)
+		return FALSE;
 
-	if (1)
+	if (!fOpRedundant)
 	{
-		GDI_RECT rect = { 0 };
-		HGDI_BRUSH brush = gdi_CreateSolidBrush(fgcolor);
+		if (!gdi_decode_color(gdi, bgcolor, &bgcolor, NULL))
+			return FALSE;
 
-		if (!brush)
+		if (!gdi_decode_color(gdi, fgcolor, &fgcolor, NULL))
 			return FALSE;
 
-		if (x > 0)
-			rect.left = x;
+		gdi_SetClipRgn(gdi->drawing->hdc, x, y, width, height);
+		gdi_SetTextColor(gdi->drawing->hdc, bgcolor);
+		gdi_SetBkColor(gdi->drawing->hdc, fgcolor);
 
-		if (y > 0)
-			rect.top = y;
+		if (1)
```

Comment:
Shouldn't we get rid of this?

Reply:
Since glyph cache is still buggy I'd suggest delaying that (does not really do anything anyway)